src/main/java/net/sf/marineapi/nmea/parser/SentenceFactory.java

```diff
@@ -69,17 +70,20 @@
  */
 public final class SentenceFactory {
 
+	// mutex to control that only one thread creates the singleton factory instance
+	private static final Semaphore INSTANCE_CREATION_MUTEX = new Semaphore(1);
+
 	// map that holds registered sentence types and parser classes
 	private static Map<String, Class<? extends SentenceParser>> parsers;
 
 	// singleton factory instance
-	private static volatile SentenceFactory instance;
+	private static SentenceFactory instance;
```

Comment:
The factory is an essential part of the library and it will most likely be used, so I think it could be created here eagerly, thus avoiding all complexity in `getInstance()`. Also, the memory footprint is quite small. What do you think, @Martin-Wegner?

Reply:
Tthat was my intention as well. I will create a new commit.